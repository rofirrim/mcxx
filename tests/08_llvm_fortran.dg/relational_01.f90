! <testinfo>
! test_generator="config/mercurium-fortran-llvm run"
! test_stdout="relational_01.f90.out"
! </testinfo>
PROGRAM MAIN
    IMPLICIT NONE
    LOGICAL :: L
    INTEGER :: X, Y, Z

    X = 1
    Y = 2
    Z = 1

    ! > 
    L = X > Y
    PRINT *, L
    L = Y > X
    PRINT *, L
    L = X > X
    PRINT *, L
    L = X > Z
    PRINT *, L

    ! <
    L = X < Y
    PRINT *, L
    L = Y < X
    PRINT *, L
    L = X < X
    PRINT *, L
    L = X < Z
    PRINT *, L

    ! <=
    L = X <= Y
    PRINT *, L
    L = Y <= X
    PRINT *, L
    L = X <= X
    PRINT *, L
    L = X <= Z
    PRINT *, L

    ! >=
    L = X >= Y
    PRINT *, L
    L = Y >= X
    PRINT *, L
    L = X >= X
    PRINT *, L
    L = X >= Z
    PRINT *, L

    ! ==
    L = X == Y
    PRINT *, L
    L = Y == X
    PRINT *, L
    L = X == X
    PRINT *, L
    L = X == Z
    PRINT *, L

    ! /=
    L = X /= Y
    PRINT *, L
    L = Y /= X
    PRINT *, L
    L = X /= X
    PRINT *, L
    L = X /= Z
    PRINT *, L
END PROGRAM MAIN
