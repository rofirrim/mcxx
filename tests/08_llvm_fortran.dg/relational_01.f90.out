 F
 T
 F
 F
 T
 F
 F
 F
 T
 F
 T
 T
 F
 T
 T
 T
 F
 F
 T
 T
 T
 T
 F
 F
